### PR TITLE
42 profile directive part3

### PR DESCRIPTION
**Pull request by dshcheung**

### CODE REVIEW

File: app/controllers/summoners_controller.rb [hunk 1]
```diff
@@ -25,20 +33,25 @@ def find_by_id
...
       else
-        render json: {success: true, summoner: response["#{name_no_space.downcase}"]}, status: 200
+        response = response["#{params['id']}"]
+        summoner = Summoner.find_by(summonerId: params['id'])
+        if summoner.nil?
+          summoner = Summoner.create(name: response['name'], profileIconId: response['profileIconId'], level: response['summonerLevel'], summonerId: response['id'], region: params['region'])
+        else 
+          summoner.update(name: response['name'], profileIconId: response['profileIconId'], level: response['summonerLevel'], summonerId: response['id'], region: params['region'])
+        end
+        render json: {success: true, summoner: summoner}, status: 200
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb:54]: You can refactor

``` ruby
        summoner = Summoner.find_by(summonerId: params['id'])
        if summoner.nil?
          summoner = Summoner.create(name: response['name'], profileIconId: response['profileIconId'], level: response['summonerLevel'], summonerId: response['id'], region: params['region'])
        else 
          summoner.update(name: response['name'], profileIconId: response['profileIconId'], level: response['summonerLevel'], summonerId: response['id'], region: params['region'])
        end
        render json: {success: true, summoner: summoner}, status: 200
```

and maybe more code

Comment (dshcheung) [inline app/controllers/summoners_controller.rb:54]: i will refactor after the profile page stuff is done as a new branch.

File: app/controllers/summoners_controller.rb [hunk 2]
```diff
@@ -62,42 +89,55 @@ def league_entry
...
-        for i in 0..response["#{params['id']}"].length - 1
-          league_name = response["#{params['id']}"][i]['queue']
-          tier = response["#{params['id']}"][i]['tier'].capitalize
-          division = response["#{params['id']}"][i]['entries'][0]['division']
+      if not response.empty?
+        response = response["#{id}"]
+        for i in 0..response.length - 1
+          league_name = response[i]['queue']
+          tier = response[i]['tier'].capitalize
+          division = response[i]['entries'][0]['division']
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb:99]: Great. Much cleaner

File: app/controllers/summoners_controller.rb [hunk 3]
```diff
@@ -147,14 +184,76 @@ def stats_summary
...
+    # stats_ranked = summoner.stats_rankeds
+    # render json: {stats_ranked: summoner[:stats_ranked]}
+    render json: {success: true}
+  end
+
+  def renew_stats_ranked
+    summoner = Summoner.find_by(summonerId: params['id'], region: params['region'])
+    update_stats_ranked(params['id'], params['region'], summoner)
+    render json: {stats_ranked: summoner[:stats_ranked]}
+  end
```
Comment (h8rry) [inline app/controllers/summoners_controller.rb:213]: Can `get_stats_ranked` be combined with `renew_stats_ranked`?

Comment (dshcheung) [inline app/controllers/summoners_controller.rb:213]: i want the user to manually update stuff unless there is no records of the user. The reason is that there is a request limit.
there is two ways to do it,
1) is the current way i have. Meaning all i need to do is to have $http use two different routes
2) is to merge these methods and have $http send in a different parameter like true/false and have the rails controller do the differentiation
i think 1) is easier and faster to implement

**Comment by h8rry** [comment]
Sounds good. Can merge.

**Pull request merged**